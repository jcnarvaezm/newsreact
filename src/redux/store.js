import { createStore } from 'redux';
import infoForm from './reducers/infoForm';

const store = createStore(infoForm);
export default store;
