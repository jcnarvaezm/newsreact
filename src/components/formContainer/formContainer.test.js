import React from 'react';
import { create, act } from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FormContainer from './formContainer';
import store from '../../redux/store';
import { Provider } from 'react-redux';
import { TextField, Checkbox } from 'emerald-ui/lib/';

configure({
  adapter: new Adapter(),
});

describe('Testing Form Container', () => {
  let component;
  let tree;
  let instance;

  beforeAll(() => {
    component = create(
      <Provider store={store}>
        <FormContainer />
      </Provider>
    );
    tree = component.toJSON();
    instance = component.root;
  });

  test('renders form container correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Content has class', () => {
    const content = instance.findByType('section');
    expect(content.props.className).toEqual('section-form');
  });
});

describe('FormContainer: Send ', () => {
  const event = {
    preventDefault: jest.fn(),
  };

  const handleSubmitFormMock = jest.fn();
  const formData = {
    firstname: 'firstname',
    lastname: 'lastname',
    email: 'email@email.com',
    phonenumber: '3216549870',
    emailtext: 'test text',
    chksendme: true,
  };

  let component;
  let form;
  let instance;
  let textInputs;
  let messageInput;

  beforeEach(() => {
    act(() => {
      component = create(
        <Provider store={store}>
          <FormContainer handleSubmitForm={handleSubmitFormMock} />
        </Provider>
      );
    });

    instance = component.root;
    form = instance.findByType('form');

    // Get form inputs
    textInputs = form.findAllByType(TextField);
    messageInput = form.findByProps({ name: 'firstname' });
  });

  it('Contact form submit without firstname', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name !== 'firstname') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: formData[input.props.name],
            },
          });
        }
      }
      messageInput.props.onChange({
        ...event,
        target: { name: 'lastname', value: 'Test message' },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit without lastname', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name !== 'lastname') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: formData[input.props.name],
            },
          });
        }
      }
      messageInput.props.onChange({
        ...event,
        target: { name: 'firtsname', value: 'Test message' },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit without email', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name !== 'email') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: formData[input.props.name],
            },
          });
        }
      }
      messageInput.props.onChange({
        ...event,
        target: { name: 'phonenumber', value: 3214567890 },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit without phonenumber', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name !== 'phonenumber') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: formData[input.props.name],
            },
          });
        }
      }
      messageInput.props.onChange({
        ...event,
        target: { name: 'email', value: 'email@email.com' },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit with phonenumber wrong', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name === 'phonenumber') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: '3214',
            },
          });
        }
      }
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit with phonenumber and email good', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name === 'phonenumber') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: formData[input.props.name],
            },
          });
        }
        if (input.props.name === 'email') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: formData[input.props.name],
            },
          });
        }
      }
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit without emailtext', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name !== 'emailtext') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: formData[input.props.name],
            },
          });
        }
      }
      messageInput.props.onChange({
        ...event,
        target: { name: 'firstname', value: 'Test message' },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit with email wrong', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        if (input.props.name === 'email') {
          input.props.onChange({
            ...event,
            target: {
              name: input.props.name,
              value: 'testfail',
            },
          });
        }
      }
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit all right', () => {
    act(() => {
      messageInput.props.onChange({
        ...event,
        target: { name: 'firstname', value: 'firstname' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'lastname', value: 'lastname' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'email', value: 'email@email.com' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'phonenumber', value: '3216549870' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'emailtext', value: 'Test text' },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit without text', () => {
    act(() => {
      for (let index = 0; index < textInputs.length; index++) {
        const input = textInputs[index];
        input.props.onChange({
          ...event,
          target: {
            name: input.props.name,
            value: '',
          },
        });
      }
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form change state checkbox', () => {
    const form = instance.findByType('form');
    const checkInput = form.findByType(Checkbox);
    act(() => {
      checkInput.props.onChange({
        ...event,
        target: {
          name: 'chksendme',
          checked: true,
        },
      });
    });
    expect(checkInput.props.value).toEqual(true);
  });

  it('Contact form submit with email and phonenumber wrong', () => {
    act(() => {
      messageInput.props.onChange({
        ...event,
        target: { name: 'firstname', value: 'firstname' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'lastname', value: 'lastname' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'email', value: 'email@email' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'phonenumber', value: '3216' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'emailtext', value: 'Test text' },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit with just email and phonenumber right', () => {
    act(() => {
      messageInput.props.onChange({
        ...event,
        target: { name: 'firstname', value: '' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'lastname', value: '' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'email', value: 'email@email.com' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'phonenumber', value: '3216549870' },
      });
      messageInput.props.onChange({
        ...event,
        target: { name: 'emailtext', value: '' },
      });
    });
    const form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleSubmitFormMock).toHaveBeenCalledTimes(0);
  });
});
