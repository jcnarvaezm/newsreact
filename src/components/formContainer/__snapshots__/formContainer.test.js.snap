// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Form Container renders form container correctly 1`] = `
<section
  className="section-form"
>
  <h2>
    Contact Us
  </h2>
  <div
    className="eui-panel container form-container eui-panel-default"
  >
    <div
      className="eui-panel-body"
    >
      <form>
        <div
          className="eui-row"
        >
          <div
            className="eui-col field-section eui-col-xs-12 eui-col-sm-6"
          >
            <div
              className="eui-text-field"
            >
              <div
                className="eui-text-field-wrapper"
              >
                <input
                  aria-errormessage="undefinedErrorMessage"
                  aria-invalid={false}
                  aria-label="First name"
                  disabled={false}
                  name="firstname"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  role="textbox"
                  type="text"
                  value=""
                />
                <div
                  className="eui-text-field-footer"
                >
                  <span
                    className="eui-text-field-message eui-text-field-error-message"
                    id="undefinedErrorMessage"
                    role="alert"
                  />
                  <span
                    className="eui-text-field-message eui-text-field-help-text"
                  />
                </div>
              </div>
              <label>
                First name
              </label>
            </div>
          </div>
          <div
            className="eui-col field-section eui-col-xs-12 eui-col-sm-6"
          >
            <div
              className="eui-text-field"
            >
              <div
                className="eui-text-field-wrapper"
              >
                <input
                  aria-errormessage="undefinedErrorMessage"
                  aria-invalid={false}
                  aria-label="Last name"
                  disabled={false}
                  name="lastname"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  role="textbox"
                  type="text"
                  value=""
                />
                <div
                  className="eui-text-field-footer"
                >
                  <span
                    className="eui-text-field-message eui-text-field-error-message"
                    id="undefinedErrorMessage"
                    role="alert"
                  />
                  <span
                    className="eui-text-field-message eui-text-field-help-text"
                  />
                </div>
              </div>
              <label>
                Last name
              </label>
            </div>
          </div>
          <div
            className="eui-col field-section eui-col-xs-12 eui-col-sm-6"
          >
            <div
              className="eui-text-field"
            >
              <div
                className="eui-text-field-wrapper"
              >
                <input
                  aria-errormessage="undefinedErrorMessage"
                  aria-invalid={false}
                  aria-label="Email"
                  disabled={false}
                  name="email"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  role="textbox"
                  type="text"
                  value=""
                />
                <div
                  className="eui-text-field-footer"
                >
                  <span
                    className="eui-text-field-message eui-text-field-error-message"
                    id="undefinedErrorMessage"
                    role="alert"
                  />
                  <span
                    className="eui-text-field-message eui-text-field-help-text"
                  />
                </div>
              </div>
              <label>
                Email
              </label>
            </div>
          </div>
          <div
            className="eui-col field-section eui-col-xs-12 eui-col-sm-6"
          >
            <div
              className="eui-text-field"
            >
              <div
                className="eui-text-field-wrapper"
              >
                <input
                  aria-errormessage="undefinedErrorMessage"
                  aria-invalid={false}
                  aria-label="Phone number"
                  disabled={false}
                  maxLength="10"
                  name="phonenumber"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  role="textbox"
                  type="number"
                  value=""
                />
                <div
                  className="eui-text-field-footer"
                >
                  <span
                    className="eui-text-field-counter"
                  >
                    0/10
                  </span>
                  <span
                    className="eui-text-field-message eui-text-field-error-message"
                    id="undefinedErrorMessage"
                    role="alert"
                  />
                  <span
                    className="eui-text-field-message eui-text-field-help-text"
                  />
                </div>
              </div>
              <label>
                Phone number
              </label>
            </div>
          </div>
          <div
            className="eui-col field-section eui-col-xs-12"
          >
            <div
              className="eui-text-field"
            >
              <div
                className="eui-text-field-wrapper"
              >
                <textarea
                  className="field-section-textarea"
                  name="emailtext"
                  onChange={[Function]}
                  value=""
                />
                <div
                  className="invalid"
                >
                  <p>
                     
                     
                  </p>
                </div>
              </div>
              <label>
                Email
              </label>
            </div>
          </div>
          <div
            className="eui-col field-section eui-col-xs-12"
          >
            <label
              className="eui-checkbox eui-has-label"
            >
              <input
                className="eui-checkbox-input"
                name="chksendme"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="checkbox"
                value={false}
              />
              <div
                className="eui-checkbox-check"
                onClick={[Function]}
                role="presentation"
              />
              <span
                className="eui-checkbox-label"
                onClick={[Function]}
                role="presentation"
              >
                Send me emails about breaking news and promotions.
              </span>
            </label>
          </div>
          <div
            className="eui-col field-section section-button eui-col-xs-12"
          >
            <button
              onClick={[Function]}
              type="submit"
            >
              Submit form
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</section>
`;
